lib/sidekiq_unique_jobs/middleware/client/strategies/unique.rb

```diff
@@ -75,7 +75,7 @@ def old_unique_for?
 
           def new_unique_for?
             connection do |conn|
-              return conn.set(payload_hash, item['at'] ? 2 : 1, nx: true, ex: expires_at)
+              return conn.set(payload_hash, item['jid'], nx: true, ex: expires_at)
```

Comment:
@mhenrixon Am I missing something? 
https://github.com/pik/sidekiq-unique-jobs/blob/master/lib/sidekiq_unique_jobs/middleware/client/strategies/unique.rb#L78
Is not identical to: https://github.com/pik/sidekiq-unique-jobs/blob/bug/unlock-mutex/lib/sidekiq_unique_jobs/middleware/client/strategies/unique.rb#L62
(Since SETNX will always fail if any value is set.) Was the idea to let simultaneous scheduling and queuing of jobs? I was under the impression this was deprecated but lacking code-comments it's hard to tell the intent? 

>  I thought the 2 and 1 is used by sidekiq to determine if it is a scheduled type of job or a regular?

As far as I understand Sidekiq just uses time['at'] to schedule into the future:   https://github.com/mperham/sidekiq/blob/master/lib%2Fsidekiq%2Fclient.rb#L186 
Where could the value Sidekiq-Unique is storing into a custom hash be accessible to Sidekiq?  

The purpose of the patch storing the `jid` as the key value is to keep Jobs from unlocking a mutex which it did not lock (if it's own key was cleared or expired).